src/test/Integration.t.sol

```diff
@@ -0,0 +1,137 @@
+
+import "../FuseFlywheelCore.sol";
+import {FlywheelStaticRewards} from "../rewards/FlywheelStaticRewards.sol";
+
+interface Comptroller {
+    function admin() external returns (address);
+
+    function _addRewardsDistributor(address distributor) external returns (uint);
+}
+
+abstract contract CErc20 is ERC20 {
+    function mint(uint256 amount) external virtual returns(uint);
+}
+
+contract FlywheelIntegrationTest is DSTestPlus {
+    FuseFlywheelCore flywheel;
+    FlywheelStaticRewards rewards;
+
+    // Pool 8 comptroller
+    Comptroller comptroller = Comptroller(0xc54172e34046c1653d1920d40333Dd358c7a1aF4);
+    
+    // fTRIBE-8
+    CErc20 fTRIBE = CErc20(0xFd3300A9a74b3250F1b2AbC12B47611171910b07);
+    
+    ERC20 tribe = ERC20(0xc7283b66Eb1EB5FB86327f08e1B5816b0720212B);
+
+    // fTRIBE-8 whale
+    address user = 0x9c5083dd4838E120Dbeac44C052179692Aa5dAC5;
+
+    // tribe treasury
+    address core = 0x8d5ED43dCa8C2F7dFB20CF7b53CC7E593635d7b9;
+
+    MockERC20 rewardToken;
+
+    function setUp() public {
+        rewardToken = new MockERC20("test token", "TKN", 18);
+        
+        flywheel = new FuseFlywheelCore(
+            rewardToken, 
+            FlywheelStaticRewards(address(0)),
+            IFlywheelBooster(address(0)),
+            address(this),
+            Authority(address(0))
+        );
+
+        rewards = new FlywheelStaticRewards(rewardToken, address(flywheel), address(this), Authority(address(0)));
+
+        flywheel.setFlywheelRewards(rewards);
+        
+        // add fTRIBE-8 to flywheel and add flywheel to the comptroller
+        flywheel.addMarketForRewards(fTRIBE);
+        hevm.prank(comptroller.admin());
+        require(comptroller._addRewardsDistributor(address(flywheel)) == 0);
+
+        // seed rewards to flywheel
+        rewardToken.mint(address(rewards), 100 ether);
+
+        // Start reward distribution at 1 token per second
+        rewards.setRewardsInfo(fTRIBE, FlywheelStaticRewards.RewardsInfo({rewardsPerSecond: 1 ether, rewardsEndTimestamp: 0}));
+
+        // advance 1 second
+        hevm.warp(block.timestamp + 1);
+    }
+
+    function testIntegration() public {
+
+        // store expected rewards per token (1 token per second over total supply)
+        uint256 rewardsPerToken = (1 ether * 1 ether) / fTRIBE.totalSupply();
+
+        // store expected user rewards (user balance times reward per second over 1 token)
+        uint256 userRewards = rewardsPerToken * fTRIBE.balanceOf(user) / 1 ether;
+        
+        // accrue rewards and check against expected
+        require(flywheel.accrue(fTRIBE, user) == userRewards);
+
+        // check market index
+        (uint224 index,) = flywheel.marketState(fTRIBE);
+        require(index == flywheel.ONE() + rewardsPerToken);
+
+        // claim and check user balance
+        flywheel.claim(user);
+        require(rewardToken.balanceOf(user) == userRewards);
+
+        // mint more tokens by user and rerun test
```

Comment:
For readability, I'd perhaps split this second test - line 91 onwards - out into a seperate integration test function

Reply:
this test kinda relies on both, I'm inclined to keep them together